### PR TITLE
feat(Orama): don't rely on static build

### PR DESCRIPTION
**Pull request by avivkeller**
Orama relies on `/next-data/api-data` and `/next-data/page-data`, which **no other part of the site needs**. This could probably be removed from the server entirely, and done locally.

Successful Orama Sync = https://github.com/nodejs/nodejs.org/actions/runs/17176946135/job/48734137782?pr=8107

### CODE REVIEW

**Comment by avivkeller** [comment]
See the new `pull_request` check running @ https://github.com/nodejs/nodejs.org/actions/runs/17169658343/job/48716786826?pr=8107

File: apps/site/scripts/orama-search/process-documents.mjs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import { slug } from 'github-slugger';
+import matter from 'gray-matter';
+
+// Note: We don't remove Markdown Headers delimiters as they're useful for delimiting sections
+// TODO(@avivkeller): Use remark
+const parseRichTextIntoPlainText = richText =>
+  richText
+    // replaces JSX and HTML and their properties with an empty string
+    // keeping only the content left
+    .replace(/<[^>]+>/gm, '')
```
[the remark this answers is not in the document]
Comment (avivkeller) [reply apps/site/scripts/orama-search/process-documents.mjs]: We are okay with this

File: .github/workflows/sync-orama.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      - main
+    types:
+      - labeled
+
+permissions:
+  contents: read
+
+jobs:
+  sync-orama-cloud:
+    name: Sync Orama Cloud
```
Comment (avivkeller) [inline .github/workflows/sync-orama.yml:26]: TODO: the if check

File: apps/site/scripts/orama-search/get-documents.mjs [hunk 3]
```diff
@@ -1,80 +1,78 @@
...
+  : undefined;
+
+/**
+ * Fetch Node.js API documentation directly from GitHub
+ * for the current Active LTS version.
+ */
+const getAPIDocs = async () => {
+  // Find the current Active LTS version
+  const releaseData = await generateReleaseData();
+  const version = releaseData.find(
```
Comment (ovflowd) [inline apps/site/scripts/orama-search/get-documents.mjs]: I know this is supposed to always return something, but can we have a constant for `release` and other for version?

File: apps/site/scripts/orama-search/get-documents.mjs [hunk 4]
```diff
@@ -1,80 +1,78 @@
...
+
+      return processDocument({
+        content: await res.text(),
+        pathname: `docs/${version}/api/${basename(name, '.md')}.html`,
+      });
+    })
   );
-}
-
-const nextPageData = readFileSync(`${dataBasePath}/page-data.body`, 'utf-8');
```
Comment (ovflowd) [inline apps/site/scripts/orama-search/get-documents.mjs:18]: The structuring of the response content seems quite different from the old version to the new one. Is this intentional?

File: apps/site/scripts/orama-search/get-documents.mjs [hunk 5]
```diff
@@ -1,80 +1,78 @@
...
+            content: await readFile(join(root, path), 'utf8'),
+            pathname: path.replace(/\.mdx?$/, ''),
+          },
+          true
+        )
+      )
+  );
+};
 
-      return {
```
Comment (ovflowd) [inline apps/site/scripts/orama-search/get-documents.mjs:72]: Like the variables that we give to Orama?

File: apps/site/scripts/orama-search/process-documents.mjs [hunk 6]
```diff
@@ -0,0 +1,86 @@
+import { sep as posixSep, basename } from 'node:path/posix';
+import { sep as winSep } from 'node:path/win32';
+
+import { slug } from 'github-slugger';
+import matter from 'gray-matter';
+
+// Note: We don't remove Markdown Headers delimiters as they're useful for delimiting sections
+// TODO(@avivkeller): Use remark
+const parseRichTextIntoPlainText = richText =>
```
Comment (ovflowd) [inline apps/site/scripts/orama-search/process-documents.mjs]: Shouldn't we use remark/unist to simply give the plain text interpretation? I mean, regexes work, but sound like quite the amount of arbitrary regexes. And IMO going simply through text nodes seems to be simpler/easier.

Comment (ovflowd) [reply apps/site/scripts/orama-search/process-documents.mjs]: Not necessarily the most performatic way, but definitely the most "stable" one.

File: apps/site/scripts/orama-search/process-documents.mjs [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+const toPosix = str => str.replaceAll(winSep, posixSep);
+
+// Derive page title from path
+const getPageTitle = path => basename(path, '.html').replace(/-/g, ' ');
+
+// Capitalize first character
+const capitalize = str => str[0].toUpperCase() + str.slice(1);
+
+// Split markdown into sections by headings
+const splitIntoSections = content => {
```
Comment (ovflowd) [inline apps/site/scripts/orama-search/process-documents.mjs]: We can also use remark for this. Almost feels like doc-kit could read our pages actually lol (wouldn't make sense to use doc-kit unorionically for this?)

**Review by ovflowd — CHANGES_REQUESTED** [review]
There are a few concerns regarding code-smell, use of more stable APIs (such as ASTs) and maybe even using our existing doc-kit for this. Also, can you please add tests for this script? I know there were none, but I actually want to verify this is working as expected from now on....

File: apps/site/scripts/orama-search/get-documents.mjs (hunk 5 above)
Comment (avivkeller) [reply apps/site/scripts/orama-search/get-documents.mjs:72]: Yes, these are passed to Orama

File: apps/site/scripts/orama-search/get-documents.mjs (hunk 4 above)
Comment (avivkeller) [reply apps/site/scripts/orama-search/get-documents.mjs:18]: The [`page-data.body` file](https://nodejs.org/en/next-data/page-data) returned an array of objects like:
```json
{
    "filename": "eol.mdx",
    "pathname": "eol",
    "title": "Node.js — End-Of-Life",
    "description": "Node.js® is a free, open-source, cross-platform JavaScript runtime environment that lets developers create servers, web apps, command line tools and scripts.",
    "content": "eJx1VktyHDcM3fcpUKVF7KrRHMC7lK3ElZKllOXPGkOiu2GxSYYge9xZ5RA5Rw6Ro+QkKZA9PSMp2XUNCeDhvQdwruDG2+v7/vqWe4JXN/e3r7urK/g6LoDewhiOcBcs7b8JJHKEQvqBZryM6z7gt5BgpiQcvEDotyBMdAq0O4iYzagfmtuS8OAxk32CIXhAiIksm4wHRyAaUhzt4UcBzj8I+JChJxTW4xxgQvYZ2QM6d6oGjj0JsIdIKVIunJcdYJ8paX6H3pNtkeTRG9J7HBTbBn6qbV0mhCM7B5JDhAOxH7bSZOGwQB4JYgrfyOR99zkOCW0FqL87zCQZbj89nPL//VcXUvczZRAyJXFeQEqMIWXoQ4Kb+9uN0u4L07GmeabGmZ0mG2Z4jzGSF/g6klL5cb14q/A/qnQkmrtbz9cSTVU9OmuxA84wEXqBrJk5t/59ABf8QMqNIZ4JSrTa3g7YG1esMrP11ESXPXwaWcCgB0c4E2CMjg3maplUvNeo4LVLobOZ5uI8JVyl3rKySCGpRjqUYQXY0NFMCQ4EPX8nu+/uAkwh0ZaIncbrobxpJHk6nhNf2HwmdJeF1k7UVJ6OlFaDVKftgGbywH1VSXB6XhD7nkyu1D/16E4jEp2ozQod/VJRbWDUEHmkaQ+fhZKAZL1tHPvhRFt4mbqiLqI3WnmyYIIl7WSUVvFAwNNEljGTW57RjTmjeRSg79EFzpqoyWNZjAtC9kmXTLLvPoXgrs2o83hIhI840JtLaDpXy4WH7OJxYoPOLYr6Ues2ei8XSmV1xEQWHB8SJiYVnRawFMlbCH4HBxfMY+UjtdrViItkmk4W3Xc3Jqw/2cR9fgMflO0YYnGYoAila9fUNopdwKYQX0zmi60Y1HWZJ9qvk+cvHd6kEgg+Bwgl27r3TiXaoKHOf99Tgj6FqfmpOrf4auXnVG/m30FfkpqodVTlPuKyptna9SFN++5tmKLjuvGUzN7hICsH7G2RnBbAYjlX0x3YQvEXWy4Vr01K3TdKxJfTirq6grdhmigZRgcPja/uHUnkTFW/cJg5FAEbjl7YUhW22fPSH7u6tROazIZ2ENKAnn9fF0WPhsAELzkpqHXwow6rz2cnQ2nbV3YgxYyAAo4GNEv1/6GVMZdU/FY40UQ+i5ZsZ/R9dRd5jVRPyx5+Cs0mAscx6ELToV3rPZmlQ8ngiawo4My+kP2fVX/x/v3HK1qRnog9xbEAzsiujeqYQhnGyvJ9JP/LA5xt/lBExTstol9TGBJOEDFlT0lGjrokm51NSYl8dst2DEfOI7ynFN7RLDqcMYWZLcGdLtrrG1/X/ao3vLq7eXhdmzql3BqKKFkRcoLQ91y7efIAjyi0PhPtHSF58Yw8UQ1FWLJ+tr8UmczodZVsLOUAI7kIh8R2aC4cMMJxZEewhFL/CehH2hRUZ2UaliZ0HUBLGdmRBfZ9SFO14g5mFq4Nndk59Xx38wARB9p3n1/YexNLL8kYirO6hGemI1k1KkKmKYaEaQEJrmi1f/74s2IP2luLQXfERTQ0hw287myTeVb7rRzo4lgl2Hf/AqMAnok="
}
```

`getAPIDocs` returns an array of objects like:
```json
  {
    content: '---\n' +
      'title: Upcoming Events\n' +
      'layout: about\n' +
      '---\n' +
      '\n' +
      '## Upcoming Events\n' +
      '\n' +
      'Node.js events are open and available to the public. Anyone is welcome to join and participate.\n' +
      '\n' +
      '### Upcoming Node.js® Meetings\n' +
      '\n' +
      'The Node.js project holds numerous meetings throughout the year to discuss and plan aspects of the project.\n' +
      '\n' +
      'The following meetings are upcoming in the next 7 days.\n' +
      '\n' +
      '<UpcomingMeetings />\n',
    pathname: 'about/get-involved/events'
  }
```

`filename` and `description` are not used, and `title` is populated during `processDocument`, with also renames the properties into the Orama-like types.

**Review by ovflowd — APPROVED** [review]
LGTM!
